sympy/core/function.py

```diff
@@ -776,9 +775,11 @@ class UndefinedFunction(FunctionClass):
     """
     The (meta)class of undefined functions.
     """
-    def __new__(mcl, name, **kwargs):
-        ret = BasicMeta.__new__(mcl, name, (AppliedUndef,), kwargs)
-        ret.__module__ = None
+    def __new__(mcl, name, bases=(AppliedUndef,), __dict__=None, **kwargs):
+        __dict__ = __dict__ or {}
+        __dict__.update(kwargs)
```

Comment:
They should be mutually exclusive. `__dict__` gets passed in by the cloudpickle unpickler, or when creating an UndefinedFunction class directly. `kwargs` gets passed in by `Function` (this lets you set attributes of the undefined function by passing them as keyword arguments to `Function`, e.g., `Function('f', is_positive=True)`).